### ISSUE TITLE
MD051 - Allow link fragments to kramdown block IAL and span IAL ids

### ISSUE THREAD

**Issue by randallpittman** [DavidAnson/markdownlint#726]
Related: #570 

kramdown lets you give ids to blocks and spans with the [inline attribute list syntax](https://kramdown.gettalong.org/syntax.html#block-ials). It would be great to not have false positives for MD051 when linking to an id of this syntax.

Example syntax

```markdown

This is the paragraph I want to link to.
{:#the-paragraph-id}

Later on, I'd like to link the [the paragraph I wrote earlier](#the-paragraph-id).

```

**Comment by DavidAnson** [comment]
Thanks for the related link. This doesn't look too bad to allow. It's funny these two systems don't even agree on the same syntax for this - tho it's pretty clear they knew about each other.
